base/docs/bootstrap.jl

```diff
@@ -15,13 +23,26 @@ macro doc(args...)
 end
 
 setexpand!() do str, obj
-    push!(docs, (current_module(), str, obj))
+    docs.head = Node((ccall(:jl_get_current_module, Any, ()), str, obj), docs.head)
     return esc(obj)
 end
 
+"""
+    DocBootstrap :: Module
+
+Basic docstring bootstrapping module that accumulates docstrings prior to the real docsystem
+being defined in `base/docs/Docs.jl`. Once the proper docsystem is loaded all docstrings
+that were stored in `DocBootstrap.docs` are migrated to their correct modules using
+`DocBootstrap.loaddocs()`.
+"""
+DocBootstrap
+
 function loaddocs()
-    for (mod, str, obj) in docs
+    node = docs.head
+    while node ≠ nothing
+        mod, str, obj = node.data
         eval(mod, :(Base.@doc($str, $obj, false)))
+        node = node.next
     end
```

Comment:
As it's currently implemented, ``DocBootstrap.docs`` is not emptied after ``loaddocs`` is run meaning that all the docs stored there are kept there unnecessarily. Should I also reset this to ``List(nothing)`` afterwards? It's been kind of useful to keep around for debugging purposes, but it's presumably wasting a bit of space.